perf/_utils.py

```diff
@@ -365,3 +365,32 @@ def pipe_cloexec():
     pipe_cloexec = os.pipe
 
     # In Python 3.4, file descriptors are non-inheritable by default (PEP 446)
+
+
+if MS_WINDOWS:
+    import msvcrt
+    if sys.version_info < (3, 0):
+        import _subprocess
+
+    class HandleOfPipe:
+        '''
+        On Windows, fd cannot be inherited but file handles can.
+        Creating inheritable handle differs in Python 2 and 3.
+        This class wraps the difference.
+
+        see also: https://digitalenginesoftware.com/blog/archives/47-Passing-pipes-to-subprocesses-in-Python-in-Windows.html
+        '''
+        def __init__(self, fd):
+            handle = msvcrt.get_osfhandle(fd)
+            if sys.version_info >= (3, 4):
+                os.set_handle_inheritable(handle, True)
+            elif sys.version_info < (3, 0):
+                curproc = _subprocess.GetCurrentProcess()
```

Comment:
hmm... I'm not familiar with Win API very well, but MSDN suggests to use CreateFile or DuplicateHandle to make inheritable handles... https://msdn.microsoft.com/en-us/library/windows/desktop/ms683463(v=vs.85).aspx

Still, os.set_handle_inheritable() sure does as you suggested, so I'll try to reproduce it in Python 2.

      DWORD flags = inheritable ? HANDLE_FLAG_INHERIT : 0;
      if (!SetHandleInformation((HANDLE)handle, HANDLE_FLAG_INHERIT, flags)) {
          ...